parchmint/device.py

```diff
@@ -34,6 +37,39 @@ def __init__(self, json=None):
             self.parse_from_json(json)
             self.generate_network()
 
+
+    # add compare function
+    # -pass device, compare devices
+    # - check connections, components, print if its same or not
+    # - have a flag to print parameter differences 
+
+    def is_feasible(self, device: Device) -> bool:
+        """compare against the input device. Return true if they are semnatcally feasible.
+
+        Args:
+            device (Device): expected device
+
+        Returns:
+            bool: If semntically feasible, return true. Else false.
+        """
+
+        self.generate_network()
+
+        FM = FeasibilityMatcher(self.G, device.G)
+
+        feasible = True
+
+        for node in self.G.nodes:
+            # find string
+            G1_component = self.get_component(node)
+            G2_component = device.get_component(node)
+
+            if not FM.semantic_feasibility(G1_component, G2_component):
+                feasible = False
+        
+        return feasible
```

Comment:
The method you're overriding will get automatically get called when you run the matcher. You don't need to iterate over each of the nodes